src/components/Documentation/RightPanel/index.tsx

```diff
@@ -34,6 +35,35 @@ const RightPanel: React.FC<IRightPanelProps> = ({
   const [currentHeadingSlug, setCurrentHeadingSlug] = useState<string | null>(
     null
   )
+
+  const generateHeadingId = (slug: string): string => `links-${slug}`
+
+  const tableOfContentsRefs: {
+    [key: string]: React.RefObject<HTMLDivElement>
+  } = headings
+    .map(({ slug }) => ({
+      id: generateHeadingId(slug),
+      ref: createRef()
+    }))
+    .reduce((acc, curr) => {
+      return {
+        ...acc,
+        [curr.id]: curr.ref
+      }
+    }, {})
```

Comment:
This dynamically creates `React ref` for us to get references to the link DOM elements.